pkg/virt-controller/ipamclaims/ipamclaims.go

```diff
@@ -0,0 +1,208 @@
+			}
+
+			err = m.ensureValidIPAMClaimForVMI(namespace, claim.Name, claim.OwnerReferences[0].UID)
+			if err != nil {
+				return fmt.Errorf("failed validating IPAMClaim %s/%s: %w", namespace, claim.Name, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func (m *IPAMClaimsManager) ensureValidIPAMClaimForVMI(namespace string, claimName string, expectedOwnerUID types.UID) error {
+	currentClaim, err := m.ipamClaimsClient.K8sV1alpha1().IPAMClaims(namespace).Get(
+		context.Background(),
+		claimName,
+		v1.GetOptions{},
+	)
+	if err != nil {
+		return fmt.Errorf("failed getting IPAMClaim: %w", err)
+	}
+
+	if len(currentClaim.OwnerReferences) != 1 || currentClaim.OwnerReferences[0].UID != expectedOwnerUID {
+		return fmt.Errorf("failed validating IPAMClaim, wrong IPAMClaim with the same name still exists")
+	}
+
+	return nil
+}
+
+func composeIPAMClaim(namespace string, ownerRef v1.OwnerReference, ipamClaimParams ipct.IPAMClaimParams, interfaceName string) *ipamclaims.IPAMClaim {
+	return &ipamclaims.IPAMClaim{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      ipamClaimParams.ClaimName,
+			Namespace: namespace,
+			OwnerReferences: []v1.OwnerReference{
+				ownerRef,
+			},
+		},
+		Spec: ipamclaims.IPAMClaimSpec{
+			Network:   ipamClaimParams.NetworkName,
+			Interface: interfaceName,
+		},
+	}
+}
+
+func (m *IPAMClaimsManager) GetNetworkToIPAMClaimParams(namespace string, vmiName string, multusNonDefaultNetworks []virtv1.Network) (map[string]ipct.IPAMClaimParams, error) {
+	nads, err := network.GetNetworkAttachmentDefinitionByName(m.networkClient.K8sCniCncfIoV1(), namespace, multusNonDefaultNetworks)
+	if err != nil {
+		return nil, fmt.Errorf("failed retrieving network attachment definitions: %w", err)
+	}
+
+	networkToIPAMClaimParams, err := ExtractNetworkToIPAMClaimParams(nads, vmiName)
+	if err != nil {
+		return nil, fmt.Errorf("failed extracting ipam claim params: %w", err)
+	}
+
+	return networkToIPAMClaimParams, nil
+}
+
+func ExtractNetworkToIPAMClaimParams(nadMap map[string]*networkv1.NetworkAttachmentDefinition, vmiName string) (map[string]ipct.IPAMClaimParams, error) {
+	networkToIPAMClaimParams := map[string]ipct.IPAMClaimParams{}
+	for networkName, nad := range nadMap {
+		persistentIPsNetworkName, err := getPersistentIPsNetworkName(nad)
+		if err != nil {
+			return nil, fmt.Errorf("failed retrieving persistentIPsNetworkName: %w", err)
+		}
+		if persistentIPsNetworkName != "" {
+			networkToIPAMClaimParams[networkName] = ipct.IPAMClaimParams{
+				ClaimName:   fmt.Sprintf("%s.%s", vmiName, networkName),
+				NetworkName: persistentIPsNetworkName,
+			}
+		}
+	}
+	return networkToIPAMClaimParams, nil
+}
+
+func getPersistentIPsNetworkName(nad *networkv1.NetworkAttachmentDefinition) (string, error) {
+	if nad.Spec.Config == "" {
+		return "", nil
+	}
+
+	netConf := struct {
+		AllowPersistentIPs bool   `json:"allowPersistentIPs,omitempty"`
+		Name               string `json:"name,omitempty"`
+	}{}
```

Comment:
```suggestion
	netConf := struct {
		Name               string `json:"name,omitempty"`
	}{}
```

Why do you need the persistent IPs information here ? I would expect to get the configuration, then the client checks the persisntent IPs value, and if enabled it proceeds to check the name. 

Not sure that makes sense given on your caller.